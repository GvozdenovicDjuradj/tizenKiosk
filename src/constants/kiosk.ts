export const SET_KIOSK_ID = "SET_KIOSK_ID"
export type SET_KIOSK_ID = typeof SET_KIOSK_ID

export const SET_KIOSK_SERIAL = "SET_KIOSK_SERIAL"
export type SET_KIOSK_SERIAL = typeof SET_KIOSK_SERIAL

export const SET_KIOSK_URL = "SET_KIOSK_URL"
export type SET_KIOSK_URL = typeof SET_KIOSK_URL

export const GET_KIOSK_ID_AND_DOMAIN_FROM_URL = "GET_KIOSK_ID_AND_DOMAIN_FROM_URL"
export type GET_KIOSK_ID_AND_DOMAIN_FROM_URL = typeof GET_KIOSK_ID_AND_DOMAIN_FROM_URL
