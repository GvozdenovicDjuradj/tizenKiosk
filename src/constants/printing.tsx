export const SET_PRINTER_CALLBACK = "SET_PRINTER_CALLBACK"
export type SET_PRINTER_CALLBACK = typeof SET_PRINTER_CALLBACK

export const PRINT_TICKET = "PRINT_TICKET"
export type PRINT_TICKET = typeof PRINT_TICKET

export const PRINT_TICKET_FULFILLED = "PRINT_TICKET_FULFILLED"
export type PRINT_TICKET_FULFILLED = typeof PRINT_TICKET_FULFILLED

export const PRINT_TICKET_REJECTED = "PRINT_TICKET_REJECTED"
export type PRINT_TICKET_REJECTED = typeof PRINT_TICKET_REJECTED
