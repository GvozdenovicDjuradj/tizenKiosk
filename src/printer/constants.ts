export const FIND_CONNECTED_PRINTERS = "FIND_CONNECTED_PRINTERS"
export type FIND_CONNECTED_PRINTERS = typeof FIND_CONNECTED_PRINTERS

export const FIND_CONNECTED_PRINTERS_FULFILLED = "FIND_CONNECTED_PRINTERS_FULFILLED"
export type FIND_CONNECTED_PRINTERS_FULFILLED = typeof FIND_CONNECTED_PRINTERS_FULFILLED

export const FIND_CONNECTED_PRINTERS_REJECTED = "FIND_CONNECTED_PRINTERS_REJECTED"
export type FIND_CONNECTED_PRINTERS_REJECTED = typeof FIND_CONNECTED_PRINTERS_REJECTED

export const SELECT_PRINTER = "SELECT_PRINTER"
export type SELECT_PRINTER = typeof SELECT_PRINTER

export const UPDATE_TICKET = "UPDATE_TICKET"
export type UPDATE_TICKET = typeof UPDATE_TICKET

export const TICKET_PRINTED = "TICKET_PRINTED"
export type TICKET_PRINTED = typeof TICKET_PRINTED

export const SET_PRINTER_ENABLED = "SET_PRINTER_ENABLED"
export type SET_PRINTER_ENABLED = typeof SET_PRINTER_ENABLED

export const CLEAR_PRINTER_DEBUG_OUTPUT = "CLEAR_PRINTER_DEBUG_OUTPUT"
export type CLEAR_PRINTER_DEBUG_OUTPUT = typeof CLEAR_PRINTER_DEBUG_OUTPUT

export const APPEND_PRINTER_DEBUG_OUTPUT = "APPEND_PRINTER_DEBUG_OUTPUT"
export type APPEND_PRINTER_DEBUG_OUTPUT = typeof APPEND_PRINTER_DEBUG_OUTPUT

export const TOGGLE_DEBUG_OUTPUT = "TOGGLE_DEBUG_OUTPUT"
export type TOGGLE_DEBUG_OUTPUT = typeof TOGGLE_DEBUG_OUTPUT

export const SET_DEBUG_OUTPUT = "SET_DEBUG_OUTPUT"
export type SET_DEBUG_OUTPUT = typeof SET_DEBUG_OUTPUT

export const DISPLAY_PRINTER_NOTIFICATION = "DISPLAY_PRINTER_NOTIFICATION"
export type DISPLAY_PRINTER_NOTIFICATION = typeof DISPLAY_PRINTER_NOTIFICATION

export const HIDE_PRINTER_NOTIFICATION = "HIDE_PRINTER_NOTIFICATION"
export type HIDE_PRINTER_NOTIFICATION = typeof HIDE_PRINTER_NOTIFICATION
