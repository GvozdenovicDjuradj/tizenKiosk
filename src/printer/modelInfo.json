{
  "models": [
    {
      "names": [
        "STAR mPOP-",
        "mPOP"
      ],
      "modelTitle": "mPOP",
      "defaultPaperWidth": 384,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "FVP10 (STR_T-001)",
        "Star FVP10"
      ],
      "modelTitle": "FVP10",
      "defaultPaperWidth": 576,
      "emulation": "StarLine"
    },
    {
      "names": [
        "TSP113",
        "TSP143",
        "TSP100-",
        "Star TSP113",
        "Star TSP143"
      ],
      "modelTitle": "TSP100",
      "defaultPaperWidth": 576,
      "emulation": "StarGraphic"
    },
    {
      "names": [
        "TSP654II (STR_T-001)",
        "TSP654 (STR_T-001)",
        "TSP651 (STR_T-001)"
      ],
      "modelTitle": "TSP650II",
      "defaultPaperWidth": 576,
      "emulation": "StarLine"
    },
    {
      "names": [
        "TSP743II (STR_T-001)",
        "TSP743 (STR_T-001)"
      ],
      "modelTitle": "TSP700II",
      "defaultPaperWidth": 576,
      "emulation": "StarLine"
    },
    {
      "names": [
        "TSP847II (STR_T-001)",
        "TSP847 (STR_T-001)"
      ],
      "modelTitle": "TSP800II",
      "defaultPaperWidth": 832,
      "emulation": "StarLine"
    },
    {
      "names": [
        "SP712 (STR-001)",
        "SP717 (STR-001)",
        "SP742 (STR-001)",
        "SP747 (STR-001)"
      ],
      "modelTitle": "SP700",
      "defaultPaperWidth": 210,
      "emulation": "StarDotImpact"
    },
    {
      "names": [],
      "modelTitle": "SM-S210i",
      "defaultPaperWidth": 384,
      "emulation": "EscPosMobile"
    },
    {
      "names": [],
      "modelTitle": "SM-S220i",
      "defaultPaperWidth": 384,
      "emulation": "EscPosMobile"
    },
    {
      "names": [],
      "modelTitle": "SM-S230i",
      "defaultPaperWidth": 384,
      "emulation": "EscPosMobile"
    },
    {
      "names": [],
      "modelTitle": "SM-T300i/T300",
      "defaultPaperWidth": 576,
      "emulation": "EscPosMobile"
    },
    {
      "names": [],
      "modelTitle": "SM-T400i",
      "defaultPaperWidth": 832,
      "emulation": "EscPosMobile"
    },
    {
      "names": [
        "STAR L200-",
        "STAR L204-"
      ],
      "modelTitle": "SM-L200",
      "defaultPaperWidth": 384,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "BSC10",
        "Star BSC10"
      ],
      "modelTitle": "BSC10",
      "defaultPaperWidth": 512,
      "emulation": "EscPos"
    },
    {
      "names": [],
      "modelTitle": "SM-S210i StarPRNT",
      "defaultPaperWidth": 384,
      "emulation": "StarPRNT"
    },
    {
      "names": [],
      "modelTitle": "SM-S220i StarPRNT",
      "defaultPaperWidth": 384,
      "emulation": "StarPRNT"
    },
    {
      "names": [],
      "modelTitle": "SM-S230i StarPRNT",
      "defaultPaperWidth": 384,
      "emulation": "StarPRNT"
    },
    {
      "names": [],
      "modelTitle": "SM-T300i StarPRNT",
      "defaultPaperWidth": 576,
      "emulation": "StarPRNT"
    },
    {
      "names": [],
      "modelTitle": "SM-T400i StarPRNT",
      "defaultPaperWidth": 832,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "STAR L300-",
        "STAR L304-"
      ],
      "modelTitle": "SM-L300",
      "defaultPaperWidth": 576,
      "emulation": "StarPRNTL"
    },
    {
      "names": [
        "MCP20 (STR-001)",
        "MCP21 (STR-001)",
        "MCP21LB",
        "mC-Print2-",
        "mC-Print2"
      ],
      "modelTitle": "mC-Print2",
      "defaultPaperWidth": 384,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "MCP31 (STR-001)",
        "mC-Print3-",
        "mC-Print3"
      ],
      "modelTitle": "mC-Print3",
      "defaultPaperWidth": 576,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "TUP592 (STR_T-001)",
        "TUP542 (STR_T-001)"
      ],
      "modelTitle": "TUP500",
      "defaultPaperWidth": 576,
      "emulation": "StarLine"
    },
    {
      "names": [
        "SK1-211_221"
      ],
      "modelTitle": "SK1-211/221/V211",
      "defaultPaperWidth": 432,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "SK1-211_221 Presenter"
      ],
      "modelTitle": "SK1-211/221/V211 Presenter",
      "defaultPaperWidth": 432,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "SK1-311_321"
      ],
      "modelTitle": "SK1-311/321/V311",
      "defaultPaperWidth": 576,
      "emulation": "StarPRNT"
    },
    {
      "names": [
        "SK1-311 Presenter"
      ],
      "modelTitle": "SK1-311/V311 Presenter",
      "defaultPaperWidth": 576,
      "emulation": "StarPRNT"
    }
  ]
}
