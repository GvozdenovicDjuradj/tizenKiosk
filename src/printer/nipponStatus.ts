import { Map } from "immutable"

export enum NipponStatus {
    SUCCESS,
    PAPER_LOW,
    PAPER_EMPTY,
    COULD_NOT_PRINT,
}

export const FriendlyMessageForStatus = Map<NipponStatus, string>([
    [NipponStatus.SUCCESS, "Success"],
    [NipponStatus.PAPER_LOW, "The paper is low"],
    [NipponStatus.PAPER_EMPTY, "The paper is empty"],
    [NipponStatus.COULD_NOT_PRINT, "Could not print a ticket"],
])

export const getFriendlyMessageForStatus = (nipponStatus: NipponStatus): string =>
    FriendlyMessageForStatus.get(nipponStatus, "Could not print a ticket")
