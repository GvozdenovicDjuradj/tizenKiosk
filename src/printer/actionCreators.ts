import { List } from "immutable"
import { Printer } from "@dnlowman/react-native-star-prnt"
import {
    APPEND_PRINTER_DEBUG_OUTPUT,
    CLEAR_PRINTER_DEBUG_OUTPUT,
    DISPLAY_PRINTER_NOTIFICATION,
    FIND_CONNECTED_PRINTERS,
    FIND_CONNECTED_PRINTERS_FULFILLED,
    FIND_CONNECTED_PRINTERS_REJECTED,
    HIDE_PRINTER_NOTIFICATION,
    SELECT_PRINTER,
    SET_DEBUG_OUTPUT,
    SET_PRINTER_ENABLED,
    TICKET_PRINTED,
    TOGGLE_DEBUG_OUTPUT,
    UPDATE_TICKET,
} from "./constants"
import { PRINT_TICKET_REJECTED } from "../constants/printing"
import { PrintTicket } from "../actions/printing"
import moment from "moment"

export interface FindConnectedPrintersAction {
    type: FIND_CONNECTED_PRINTERS;
}

export const findConnectedPrinters = (): FindConnectedPrintersAction => ({
    type: FIND_CONNECTED_PRINTERS
})

export interface FindConnectedPrintersFulfilledAction {
    type: FIND_CONNECTED_PRINTERS_FULFILLED;
    payload: {
        printers: List<Printer>;
    };
}

export const findConnectedPrintersFulfilled = (printers: List<Printer>): FindConnectedPrintersFulfilledAction => ({
    type: FIND_CONNECTED_PRINTERS_FULFILLED,
    payload: {
        printers
    }
})

export interface FindConnectedPrintersRejectedAction {
    type: FIND_CONNECTED_PRINTERS_REJECTED;
}

export const findConnectedPrintersRejected = (): FindConnectedPrintersRejectedAction => ({
    type: FIND_CONNECTED_PRINTERS_REJECTED
})

export interface SelectPrinterAction {
    type: SELECT_PRINTER;
    payload: {
        printer: Printer;
    }
}

export const selectPrinter = (printer: Printer): SelectPrinterAction => ({
    type: SELECT_PRINTER,
    payload: {
        printer
    }
})

export interface UpdateTicketAction {
    type: UPDATE_TICKET;
    payload: {
        logoUrl: string;
        ticketMessage: string;
        showAtTopValue: string;
        localCurrentTime: string;
        localCurrentDate: string;
        logoFooterUrl: string;
    }
}

export const updateTicket =
    (logoUrl: string, ticketMessage: string, showAtTopValue: string,
        localCurrentTime: string, localCurrentDate: string, logoFooterUrl: string): UpdateTicketAction => ({
            type: UPDATE_TICKET,
            payload: {
                logoUrl,
                ticketMessage,
                showAtTopValue,
                localCurrentTime,
                localCurrentDate,
                logoFooterUrl
            }
        })

export interface TicketPrintedAction {
    type: TICKET_PRINTED;
}

export const ticketPrinted = (): TicketPrintedAction => ({
    type: TICKET_PRINTED
})

export interface PrintTicketRejectedAction {
    type: PRINT_TICKET_REJECTED;
    payload: {
        title: string;
        message: string;
    }
}

export const printTicketRejected = (): PrintTicketRejectedAction => ({
    type: PRINT_TICKET_REJECTED,
    payload: {
        title: "Ticket could not be printed",
        message: "We could not print a ticket for you please remember your queue position."
    }
})

export interface SetPrinterEnabledAction {
    type: SET_PRINTER_ENABLED,
    payload: {
        enabled: boolean
    }
}

export const setPrinterEnabled = (enabled: boolean): SetPrinterEnabledAction => ({
    payload: {
        enabled
    },
    type: SET_PRINTER_ENABLED,
})

export interface ClearPrinterDebugOutput {
    type: CLEAR_PRINTER_DEBUG_OUTPUT
}

export const clearPrinterDebugOutput = (): ClearPrinterDebugOutput => ({
    type: CLEAR_PRINTER_DEBUG_OUTPUT
})

export interface AppendPrinterDebugOutput {
    type: APPEND_PRINTER_DEBUG_OUTPUT,
    payload: {
        messages: string[]
    }
}

export const appendPrinterDebugOutput = (...messages: string[]): AppendPrinterDebugOutput => ({
    type: APPEND_PRINTER_DEBUG_OUTPUT,
    payload: {
        messages: messages.map((message) => `[${moment().toISOString()}] - ${message}`)
    }
})

export interface ToggleDebugOutputAction {
    type: TOGGLE_DEBUG_OUTPUT;
}

export const toggleDebugOutput = (): ToggleDebugOutputAction => ({
    type: TOGGLE_DEBUG_OUTPUT
})

export interface SetDebugOutputAction {
    type: SET_DEBUG_OUTPUT;
    payload: {
        value: boolean;
    }
}

export const setDebugOutput = (value: boolean): SetDebugOutputAction => ({
    type: SET_DEBUG_OUTPUT,
    payload: {
        value
    }
})

export interface DisplayPrinterNotificationAction {
    type: DISPLAY_PRINTER_NOTIFICATION;
    payload: {
        title: string;
        message: string;
    };
}

export const displayPrinterNotification = (title: string, message: string): DisplayPrinterNotificationAction => ({
    type: DISPLAY_PRINTER_NOTIFICATION,
    payload: {
        title,
        message
    }
})

export interface HidePrinterNotificationAction {
    type: HIDE_PRINTER_NOTIFICATION
}

export const hidePrinterNotification = (): HidePrinterNotificationAction => ({
    type: HIDE_PRINTER_NOTIFICATION
})

export type PrinterActions = FindConnectedPrintersAction
    | FindConnectedPrintersFulfilledAction
    | FindConnectedPrintersRejectedAction
    | SelectPrinterAction
    | UpdateTicketAction
    | TicketPrintedAction
    | PrintTicketRejectedAction
    | SetPrinterEnabledAction
    | ClearPrinterDebugOutput
    | AppendPrinterDebugOutput
    | ToggleDebugOutputAction
    | DisplayPrinterNotificationAction
    | HidePrinterNotificationAction
    | SetDebugOutputAction
    | PrintTicket
