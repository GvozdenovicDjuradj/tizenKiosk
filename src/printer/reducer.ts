import { List } from "immutable"
import { Printer } from "@dnlowman/react-native-star-prnt"
import {
    APPEND_PRINTER_DEBUG_OUTPUT,
    CLEAR_PRINTER_DEBUG_OUTPUT,
    DISPLAY_PRINTER_NOTIFICATION,
    FIND_CONNECTED_PRINTERS,
    FIND_CONNECTED_PRINTERS_FULFILLED,
    FIND_CONNECTED_PRINTERS_REJECTED,
    HIDE_PRINTER_NOTIFICATION,
    SELECT_PRINTER,
    SET_DEBUG_OUTPUT,
    SET_PRINTER_ENABLED,
    TICKET_PRINTED,
    TOGGLE_DEBUG_OUTPUT,
    UPDATE_TICKET
} from "./constants"
import { PrinterActions } from "./actionCreators"
import { Reducer } from "redux"
import { APP } from "../actions/types"
import { PRINT_TICKET } from "../constants/printing"

const { freeze } = Object

export interface PrinterState {
    isFindingPrinters: boolean;
    connectedPrinters: List<Printer>;
    selectedPrinter?: Printer;
    logoUrl: string;
    ticketMessage: string;
    showAtTopValue: string;
    localCurrentTime: string;
    localCurrentDate: string;
    logoFooterUrl: string;
    isPendingPrint: boolean;
    isPrinterEnabled: boolean;
    debugOutput: string[];
    isDebugOutputVisible: boolean;
    notificationTitle: string;
    notificationMessage: string;
    isNotificationVisible: boolean;
    isCurrentPrintATest: boolean;
}

export const initialState: Readonly<PrinterState> = freeze({
    isFindingPrinters: false,
    connectedPrinters: List<Printer>(),
    selectedPrinter: undefined,
    logoUrl: "",
    ticketMessage: "",
    showAtTopValue: "",
    localCurrentTime: "",
    localCurrentDate: "",
    logoFooterUrl: "",
    isPendingPrint: false,
    isPrinterEnabled: false,
    debugOutput: [],
    isDebugOutputVisible: false,
    notificationTitle: "",
    notificationMessage: "",
    isNotificationVisible: false,
    isCurrentPrintATest: false,
})

const reducer: Reducer<PrinterState, PrinterActions> =
    (state: PrinterState = initialState, action: PrinterActions): Readonly<PrinterState> => {
        switch (action.type) {
            case PRINT_TICKET:
                return {
                    ...state,
                    isCurrentPrintATest: action.payload.isTest
                }

            case APP.SECRET_TAP.SUCCESS:
                return {
                    ...state,
                    isPendingPrint: false
                }

            case FIND_CONNECTED_PRINTERS:
                return {
                    ...state,
                    connectedPrinters: List.of<Printer>(),
                    isFindingPrinters: true,
                    selectedPrinter: undefined,
                }

            case FIND_CONNECTED_PRINTERS_FULFILLED:
                return {
                    ...state,
                    connectedPrinters: action.payload.printers,
                    isFindingPrinters: false,
                }

            case FIND_CONNECTED_PRINTERS_REJECTED:
                return {
                    ...state,
                    isFindingPrinters: false
                }

            case SELECT_PRINTER:
                return {
                    ...state,
                    selectedPrinter: action.payload.printer,
                }

            case UPDATE_TICKET:
                return {
                    ...state,
                    logoUrl: action.payload.logoUrl,
                    ticketMessage: action.payload.ticketMessage,
                    showAtTopValue: action.payload.showAtTopValue,
                    localCurrentTime: action.payload.localCurrentTime,
                    localCurrentDate: action.payload.localCurrentDate,
                    logoFooterUrl: action.payload.logoFooterUrl,
                    isPendingPrint: true
                }

            case TICKET_PRINTED:
                return {
                    ...state,
                    isPendingPrint: false,
                    isNotificationVisible: false,
                    notificationTitle: "",
                    notificationMessage: "",
                }

            case SET_PRINTER_ENABLED:
                return {
                    ...state,
                    isPrinterEnabled: action.payload.enabled,
                    selectedPrinter: initialState.selectedPrinter,
                }

            case CLEAR_PRINTER_DEBUG_OUTPUT:
                return {
                    ...state,
                    debugOutput: []
                }

            case APPEND_PRINTER_DEBUG_OUTPUT:
                return {
                    ...state,
                    debugOutput: [...state.debugOutput, ...action.payload.messages]
                }

            case TOGGLE_DEBUG_OUTPUT:
                return {
                    ...state,
                    isDebugOutputVisible: !state.isDebugOutputVisible
                }

            case DISPLAY_PRINTER_NOTIFICATION:
                return {
                    ...state,
                    notificationTitle: action.payload.title,
                    notificationMessage: action.payload.message,
                    isNotificationVisible: __DEV__,
                    // This functionality is currently only available on a dev build as we still need to implement i18n
                }

            case HIDE_PRINTER_NOTIFICATION:
                return {
                    ...state,
                    isNotificationVisible: false,
                }

            case SET_DEBUG_OUTPUT:
                return {
                    ...state,
                    isDebugOutputVisible: action.payload.value
                }

            default:
                return state
        }
    }

export default reducer
