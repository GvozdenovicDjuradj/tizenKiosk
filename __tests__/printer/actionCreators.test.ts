import { List } from "immutable"
import { Printer } from "@dnlowman/react-native-star-prnt"
import {
    appendPrinterDebugOutput,
    clearPrinterDebugOutput,
    findConnectedPrinters,
    findConnectedPrintersFulfilled,
    findConnectedPrintersRejected,
    selectPrinter,
    setPrinterEnabled,
    toggleDebugOutput,
    displayPrinterNotification,
    hidePrinterNotification, setDebugOutput,
} from "../../src/printer/actionCreators"
import { APPEND_PRINTER_DEBUG_OUTPUT } from "../../src/printer/constants"

describe(`the printer actionCreators module`, () => {
    it(`should have an action creator called findConnectedPrinters which returns the correct action`, () => {
        // When
        const action = findConnectedPrinters()

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called findConnectedPrintersFulfilled which returns the correct action`, () => {
        // Given
        const printers: List<Printer> = List.of({
            modelName: "Example",
            macAddress: "Example",
            portName: "Example",
            USBSerialNumber: "Example"
        })

        // When
        const action = findConnectedPrintersFulfilled(printers)

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called findConnectedPrintersRejected which returns the correct action`, () => {
        // When
        const action = findConnectedPrintersRejected()

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called selectPrinter which returns the correct action`, () => {
        // Given
        const printer: Printer = {
            modelName: "Test",
            portName: "Test"
        }

        // When
        const action = selectPrinter(printer)

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called setPrinterEnabled which returns the correct action`, () => {
        // Given
        const enabled = true

        // When
        const action = setPrinterEnabled(enabled)

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called clearPrinterDebugOutput which returns the correct action`, () => {
        // When
        const action = clearPrinterDebugOutput()

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called appendPrinterDebugOutput which returns the correct action`, () => {
        // Given
        const message = "SOME EXAMPLE PRINTER OUTPUT"

        // When
        const action = appendPrinterDebugOutput(message)

        // Then
        expect(action.type).toBe(APPEND_PRINTER_DEBUG_OUTPUT)
        expect(action.payload.messages).toHaveLength(1)
        expect(action.payload.messages[0]).toContain("SOME EXAMPLE PRINTER OUTPUT")
    })

    it(`should have an action creator called appendPrinterDebugOutput
        which supports taking an array of strings`, () => {
        // Given
        const messages = ["Hello", "World"]

        // When
        const action = appendPrinterDebugOutput(...messages)

        // Then
        expect(action).toBeDefined()
        expect(action).toHaveProperty("type", APPEND_PRINTER_DEBUG_OUTPUT)
        expect(action.payload.messages).toHaveLength(2)
        expect(action.payload.messages[0]).toContain(messages[0])
        expect(action.payload.messages[1]).toContain(messages[1])
    })

    it(`should have an action creator called toggleDebugOutput which returns the correct action`, () => {
        // When
        const action = toggleDebugOutput()

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called displayPrinterNotification which returns the correct action`, () => {
        // Given
        const title = "example title"
        const message = "example message"

        // When
        const action = displayPrinterNotification(title, message)

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should have an action creator called hidePrinterNotification which returns the correct action`, () => {
        // When
        const action = hidePrinterNotification()

        // Then
        expect(action).toMatchSnapshot()
    })

    it(`should export setDebugOutput which returns the correct action`, () => {
        // When
        const action = setDebugOutput(true)

        // Then
        expect(action).toMatchSnapshot()
    })
})
