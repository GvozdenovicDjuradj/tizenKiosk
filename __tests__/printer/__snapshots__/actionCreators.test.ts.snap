// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the printer actionCreators module should export setDebugOutput which returns the correct action 1`] = `
Object {
  "payload": Object {
    "value": true,
  },
  "type": "SET_DEBUG_OUTPUT",
}
`;

exports[`the printer actionCreators module should have an action creator called clearPrinterDebugOutput which returns the correct action 1`] = `
Object {
  "type": "CLEAR_PRINTER_DEBUG_OUTPUT",
}
`;

exports[`the printer actionCreators module should have an action creator called displayPrinterNotification which returns the correct action 1`] = `
Object {
  "payload": Object {
    "message": "example message",
    "title": "example title",
  },
  "type": "DISPLAY_PRINTER_NOTIFICATION",
}
`;

exports[`the printer actionCreators module should have an action creator called findConnectedPrinters which returns the correct action 1`] = `
Object {
  "type": "FIND_CONNECTED_PRINTERS",
}
`;

exports[`the printer actionCreators module should have an action creator called findConnectedPrintersFulfilled which returns the correct action 1`] = `
Object {
  "payload": Object {
    "printers": Immutable.List [
      Object {
        "USBSerialNumber": "Example",
        "macAddress": "Example",
        "modelName": "Example",
        "portName": "Example",
      },
    ],
  },
  "type": "FIND_CONNECTED_PRINTERS_FULFILLED",
}
`;

exports[`the printer actionCreators module should have an action creator called findConnectedPrintersRejected which returns the correct action 1`] = `
Object {
  "type": "FIND_CONNECTED_PRINTERS_REJECTED",
}
`;

exports[`the printer actionCreators module should have an action creator called hidePrinterNotification which returns the correct action 1`] = `
Object {
  "type": "HIDE_PRINTER_NOTIFICATION",
}
`;

exports[`the printer actionCreators module should have an action creator called selectPrinter which returns the correct action 1`] = `
Object {
  "payload": Object {
    "printer": Object {
      "modelName": "Test",
      "portName": "Test",
    },
  },
  "type": "SELECT_PRINTER",
}
`;

exports[`the printer actionCreators module should have an action creator called setPrinterEnabled which returns the correct action 1`] = `
Object {
  "payload": Object {
    "enabled": true,
  },
  "type": "SET_PRINTER_ENABLED",
}
`;

exports[`the printer actionCreators module should have an action creator called toggleDebugOutput which returns the correct action 1`] = `
Object {
  "type": "TOGGLE_DEBUG_OUTPUT",
}
`;
